service/controller/clusterapi/v27/adapter/guest_auto_scaling_group.go

```diff
@@ -47,8 +47,8 @@ func (a *GuestAutoScalingGroupAdapter) Adapt(cfg Config) error {
 	a.ASGDesiredCapacity = currentDesiredMinWorkers
 	a.ASGMaxSize = maxWorkers
 	a.ASGMinSize = minWorkers
-	a.ASGType = legacykey.KindWorker
-	a.ClusterID = legacykey.ClusterID(cfg.CustomObject)
+	a.ASGType = "worker"
```

Comment:
Why did you get rid of the `KindWorker` and other `Kind*` keys? I mean not really necessary for master and worker but there is also etcd-elb and so on below.